solidity/contracts/bridge/Bridge.sol

```diff
@@ -141,36 +180,165 @@ contract Bridge {
+            address depositorAddress =
+                scriptSignature.toAddress(scriptSignature.length - 91);
+
+            uint256 depositId =
+                uint256(
+                    keccak256(
+                        abi.encode(
+                            fundingTxHash,
+                            fundingOutputIndex,
+                            depositorAddress
+                        )
+                    )
+                );
+
+            // TODO We need to validate if the sum in the output minus the
+            //      amount from the previous wallet balance input minus fees is
+            //      equal to the amount by which Bank balances were increased.
+            //
+            // TODO We need to validate txOutput to see if the balance was not
+            //      transferred away from the wallet before increasing balances in
+            //      the bank.
+            //
+            // TODO Delete deposit from unswept mapping or mark it as swept
+            //      depending on the gas costs. Alternatively, do not allow to
+            //      use the same TX input vector twice. Sweep should be provable
+            //      only one time.
+        }
+    }
+
+    function evaluateProofDifficulty(bytes memory bitcoinHeaders) public view {
+        uint256 reqDiff;
+        uint256 current = relay.getCurrentEpochDifficulty();
+        uint256 previous = relay.getPrevEpochDifficulty();
+        uint256 firstHeaderDiff =
+            bitcoinHeaders.extractTarget().calculateDifficulty();
+
+        if (firstHeaderDiff == current) {
+            reqDiff = current;
+        } else if (firstHeaderDiff == previous) {
+            reqDiff = previous;
+        } else {
+            revert("not at current or previous difficulty");
+        }
+
+        uint256 observedDiff = bitcoinHeaders.validateHeaderChain();
+
+        require(
+            observedDiff != ValidateSPV.getErrBadLength(),
+            "Invalid length of the headers chain"
+        );
+        require(
+            observedDiff != ValidateSPV.getErrInvalidChain(),
+            "Invalid headers chain"
+        );
+        require(
+            observedDiff != ValidateSPV.getErrLowWork(),
+            "Insufficient work in a header"
+        );
+
+        require(
+            observedDiff >= reqDiff * TX_PROOF_DIFFICULTY_FACTOR,
+            "Insufficient accumulated difficulty in header chain"
+        );
+    }
+
+    function checkProofFromTxId(
+        bytes32 txId,
+        bytes memory merkleProof,
+        uint256 txIndexInBlock,
+        bytes memory bitcoinHeaders
+    ) internal {
+        require(
+            txId.prove(
+                bitcoinHeaders.extractMerkleRootLE().toBytes32(),
+                merkleProof,
+                txIndexInBlock
+            ),
+            "Tx merkle proof is not valid for provided header and txId"
+        );
+        evaluateProofDifficulty(bitcoinHeaders);
+    }
+
+    function validateSweepProof(
```

Comment:
Can we put `checkProofFromTxId` and `evaluateProofDifficulty` below `validateSweepProof` to have the code organized in the order it is called? I think this will improve readability since we usually browse from top to bottom.

Reply:
Done. Although I removed `validateSweepProof` (I did some refactoring).